### ISSUE TITLE
brunch missing during "vagrant up test"

### ISSUE THREAD

**Issue by kdemarest** [HubSpot/Singularity#552]
During a "vagrant up test", while building SingularityService, the module SingularityUI is built. In SingularityUI's package.json file, the postinstall script calls brunch but returns the following error:
```
==> test: [INFO] > bower install --allow-root && brunch build --production
==> test: [ERROR] sh: 1: brunch: not found
```
However, checking the package.json everything appears fine (see below) and when I `vagrant ssh` into the machine, the node_modules/brunch/bin/ directory contains brunch.

Here is the excerpt from the package.json for reference:

```
"bin": {
    "bower": "./node_modules/bower/bin/bower",
    "brunch": "./node_modules/brunch/bin/brunch"
  },
  "scripts": {
    "start": "brunch watch --server",
    "postinstall": "bower install --allow-root && brunch build --production"
  },
  "dependencies": {
    "bower": "~1.4.1",
    "brunch": "^1.7.14",
```

Here is the full output for reference:

```
==> test: Downloaded: https://repo.maven.apache.org/maven2/com/google/guava/guava/10.0.1/guava-10.0.1.jar (1467 KB at 290.1 KB/sec)
==> test: [INFO] Node v0.10.28 is already installed.
==> test: [INFO] Found NPM version 1.4.9
==> test: [INFO] 
==> test: [INFO] --- frontend-maven-plugin:0.0.23:npm (npm install) @ SingularityService ---
==> test: [INFO] Running 'npm install --color=false' in /singularity/SingularityUI
==> test: [INFO] 
==> test: [INFO] > SingularityUI@0.3.0 postinstall /singularity/SingularityUI
==> test: [INFO] > bower install --allow-root && brunch build --production
==> test: [INFO] 
==> test: [ERROR] sh: 1: brunch: not found
==> test: [ERROR] 
==> test: [ERROR] npm ERR! SingularityUI@0.3.0 postinstall: `bower install --allow-root && brunch build --production`
==> test: [ERROR] npm ERR! Exit status 127
==> test: [ERROR] npm ERR! 
==> test: [ERROR] npm ERR! Failed at the SingularityUI@0.3.0 postinstall script.
==> test: [ERROR] npm ERR! This is most likely a problem with the SingularityUI package,
==> test: [ERROR] npm ERR! not with npm itself.
==> test: [ERROR] npm ERR! Tell the author that this fails on your system:
==> test: [ERROR] npm ERR!     bower install --allow-root && brunch build --production
==> test: [ERROR] npm ERR! You can get their info via:
==> test: [ERROR] npm ERR!     npm owner ls SingularityUI
==> test: [ERROR] npm ERR! There is likely additional logging output above.
==> test: [ERROR] npm ERR! System Linux 3.13.0-24-generic
==> test: [ERROR] npm ERR! command "/singularity/SingularityUI/node/node" "/singularity/SingularityUI/node/npm/bin/npm-cli.js" "install" "--color=false"
==> test: [ERROR] npm ERR! cwd /singularity/SingularityUI
==> test: [ERROR] npm ERR! node -v v0.10.28
==> test: [ERROR] npm ERR! npm -v 1.4.9
==> test: [ERROR] npm ERR! code ELIFECYCLE
==> test: [ERROR] npm ERR! 
==> test: [ERROR] npm ERR! Additional logging details can be found in:
==> test: [ERROR] npm ERR!     /singularity/SingularityUI/npm-debug.log
==> test: [ERROR] npm ERR! not ok code 0
==> test: [INFO] ------------------------------------------------------------------------
==> test: [INFO] Reactor Summary:
==> test: [INFO] 
==> test: [INFO] Singularity ........................................ SUCCESS [01:45 min]
==> test: [INFO] SingularityBase .................................... SUCCESS [ 24.655 s]
==> test: [INFO] SingularityMesosClient ............................. SUCCESS [  6.040 s]
==> test: [INFO] SingularitySwagger ................................. SUCCESS [  7.896 s]
==> test: [INFO] SingularityService ................................. FAILURE [01:09 min]
==> test: [INFO] SingularityRunnerBase .............................. SKIPPED
==> test: [INFO] SingularityS3Base .................................. SKIPPED
==> test: [INFO] SingularityClient .................................. SKIPPED
==> test: [INFO] SingularityExecutor ................................ SKIPPED
==> test: [INFO] SingularityExecutorCleanup ......................... SKIPPED
==> test: [INFO] SingularityLogWatcher .............................. SKIPPED
==> test: [INFO] SingularityS3Uploader .............................. SKIPPED
==> test: [INFO] SingularityS3Downloader ............................ SKIPPED
==> test: [INFO] SingularityOOMKiller ............................... SKIPPED
==> test: [INFO] EmbedSingularityExample ............................ SKIPPED
==> test: [INFO] ------------------------------------------------------------------------
==> test: [INFO] BUILD FAILURE
==> test: [INFO] ------------------------------------------------------------------------
==> test: [INFO] Total time: 03:44 min
==> test: [INFO] Finished at: 2015-05-18T02:21:19+00:00
==> test: [INFO] Final Memory: 42M/100M
==> test: [INFO] ------------------------------------------------------------------------
==> test: [ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:0.0.23:npm (npm install) on project SingularityService: Failed to run task: 'npm install --color=false' failed. (error code 1) -> [Help 1]
==> test: [ERROR] 
==> test: 
==> test: [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
==> test: [ERROR] Re-run Maven using the -X switch to enable full debug logging.
==> test: [ERROR] 
==> test: [ERROR] For more information about the errors and possible solutions, please read the following articles:
==> test: [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
==> test: [ERROR] 
==> test: [ERROR] After correcting the problems, you can resume the build with the command
==> test: [ERROR]   mvn <goals> -rf :SingularityService
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

**Comment by tpetr** [comment]
strangely, brunch installs fine for me (from a `vagrant destroy test && vagrant up test`'):

```
==> test: [INFO] Node v0.10.28 is already installed.
==> test: [INFO] Found NPM version 1.4.9
==> test: [INFO]
==> test: [INFO] --- frontend-maven-plugin:0.0.23:npm (npm install) @ SingularityService ---
==> test: [INFO] Running 'npm install --color=false' in /singularity/SingularityUI
==> test: [INFO]
==> test: [INFO] > SingularityUI@0.3.0 postinstall /singularity/SingularityUI
==> test: [INFO] > bower install --allow-root && brunch build --production
==> test: [INFO]
==> test: [INFO] 18 May 13:25:54 - info: compiled 189 files into 2 files, copied 8 in 10577ms
==> test: [INFO]
==> test: [INFO] --- build-helper-maven-plugin:1.9.1:add-resource (add-generated-resources) @ SingularityService ---
```

**Comment by kdemarest** [comment]
No longer reproducible after the fix to #550.
